pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -205,27 +205,25 @@ type ClusterWorkspaceTypeSpec struct {
 	// +optional
 	DefaultChildWorkspaceType *ClusterWorkspaceTypeReference `json:"defaultChildWorkspaceType,omitempty"`
 
-	// allowAnyChildWorkspaceTypes permits all ClusterWorkspaceTypes to be
-	// created in a workspace of this type.
-	//
-	// +optional
-	AllowAnyChildWorkspaceTypes bool `json:"allowAnyChildWorkspaceTypes,omitempty"`
-
-	// allowedChildWorkspaceTypes is a list of ClusterWorkspaceTypes that can be
-	// created in a workspace of this type.
+	// allowedChildren is a set of ClusterWorkspaceTypes that can be created in a
+	// workspace of this type.
 	//
 	// By default, no type is allowed. This means no other workspace can be nested
-	// within a workspace of the given type. Use allowAnyChildWorkspaceTypes to be
+	// within a workspace of the given type. Use allowedChildren.any=true to be
 	// permissive with child types.
 	//
 	// +optional
-	AllowedChildWorkspaceTypes []ClusterWorkspaceTypeReference `json:"allowedChildWorkspaceTypes,omitempty"`
+	AllowedChildren *ClusterWorkspaceTypeSelector `json:"allowedChildren,omitempty"`
```

Comment:
can we switch the default? We should be permissive in my eyes, and you nice new types allow that. OpenAPI can do.

Reply:
same below